### PR TITLE
Menu upgrade

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?
Menus are managed by `Gtk.UIManager`, which is a deprecated API. This API will be removed in GTK4.

Issue Number: #202 

### What is the new behavior?

The menu uses the CSD structure (icons in header bar), which makes for a clean design.

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

* All plugins should publish their menu entries in a not yet defined way. The menu is less flexible, so I probably will provide a few extension points.

### Other information

* #202 contains some references.
* A new service (RecentFiles) is introduced, so you'll have to run a `poetry install` before Gaphor runs from this branch.

## To do

- [ ] Keyboard shortcuts for toolbox items

### CODE REVIEW

**Comment by amolenaar** [comment]
Work in progress:

image

**Comment by danyeaw** [comment]
@amolenaar I still have a file menu when I tried it locally, but I don't see that in your screenshot. Are you planning on removing that as well? This is coming together, the new header bar looks great!

**Comment by amolenaar** [comment]
I temporarily removed the menu for the screenshot ;), so you would have a picture of what it will look like in the end. I still want the menu around, so I know which items still need to be migrated.

**Comment by amolenaar** [comment]
I integrated the editor with the main window.

image

**Comment by danyeaw** [comment]
@amolenaar I just gave this a spin, this looks really good! I like the new element editor on the sidebar, and how it hides everything not essential doing some basic modeling.

**Comment by amolenaar** [comment]
Thanks. Almost all functionality is in. Some plugins (python import, alignment) need fixing anyway, so I can leave them out for now and add them once they work again.

**Comment by danyeaw** [comment]
@amolenaar I am getting an AttributeError when creating a block following your latest commits:
```
Traceback (most recent call last):
  File "/home/dan/Projects/gaphor/gaphor/ui/diagrampage.py", line 284, in _on_view_selection_changed
    DiagramSelectionChange(view, view.focused_item, view.selected_items)
  File "/home/dan/Projects/gaphor/gaphor/services/eventmanager.py", line 60, in handle
    self._events.handle(e)
  File "/home/dan/Projects/gaphor/gaphor/misc/generic/event.py", line 62, in handle
    handler(event)
  File "/home/dan/Projects/gaphor/gaphor/services/copyservice.py", line 48, in _update
    diagram_view.get_action_group("win").lookup_action("edit-copy").set_enabled(
AttributeError: 'NoneType' object has no attribute 'lookup_action'
```

File: gaphor/plugins/diagramexport/tests/test_diagramexport.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@pytest.fixture
+def canvas():
+    c = gaphas.Canvas()
+    c.add(gaphas.examples.Box())
+    return c
+
+
+def test_export_to_svg(diagram_export, canvas, tmp_path):
```
Comment (danyeaw) [inline gaphor/plugins/diagramexport/tests/test_diagramexport.py]: Nice, I didn't know that pytest had a temp path fixture builtin!

**Comment by amolenaar** [comment]
> ```
> AttributeError: 'NoneType' object has no attribute 'lookup_action'
> ```

Thanks. I'll check it out. I find it somewhat hard to build unit tests around this code, since it hooks deeply into Gtk. At some points I'm dependent on running the Gtk main loop. (which is possible, but I'd rather not do)

**Comment by danyeaw** [comment]
@amolenaar Ya, GUI testing is hard. We could use something PyAutoGUI, but I'm not sure it would be worth the effort.

**Comment by amolenaar** [comment]
I took the liberty of merging this branch. There has been quite some change and a couple of fixes.

I would like to do the GUI automation just below the GUI. With actions this is possible and out application startup times are pretty good, so  we can start a "complete" application just for a single test. 

BTW. I was not able to reproduce the error you mentioned before.

**Pull request merged**